### DISCUSSION TITLE
Allow finding and focusing terminal panes using command-palette search

### DISCUSSION THREAD

**Discussion by Blankeos** [zed-industries/zed#23945] [Feature Requests]
The `Cmd + P` keybind allows searching for files and opening/focusing them, but that doesn't work with and there is no equivalent workflow that allows searching for the name of a terminal pane and focusing it.

Here's a demonstration of 
1) Opening a terminal pane
2) Searching using `Cmd+P` to open files
3) Trying to search for the open terminal pane using `Cmd+P` and failing

![Zed Search for Screen Terminal](https://github.com/zed-industries/zed/assets/38070918/67327a38-d162-41b8-82f6-51f4dbfdfc85)

**Comment by JosephTLyons** [comment]
There isn't a way to currently do that, but it seems like a neat idea!

**Comment by alygin** [comment]
[Tab Swtitcher](https://github.com/zed-industries/zed/issues/) will do the job for simple (the most common?) cases &mdash; [switching](https://github.com/zed-industries/zed/pull/7987#issue-2141161429) to a recently used terminal _in the same pane_.

Though it will probably not have searching capabilities.

**Comment by probably-neb** [comment]
Hey! Thanks for bringing this to our attention it's definitely an interesting idea. We've decided to start moving feature requests like this to discussions. Feel free to continue discussing there.

**Comment by P1n3appl3** [comment]
#28047 landed recently which adds a new picker that does include terminals. The only exception is that terminals in the dock aren't included right now, only ones in center panes. I think this mostly addresses the feature request so I'll close this thread.
